azure/services/identities/client.go

```diff
@@ -39,7 +39,7 @@ type AzureClient struct {
 }
 
 // NewClient creates a new MSI client from an authorizer.
-func NewClient(auth azure.Authorizer) (*AzureClient, error) {
+func NewClient(auth azure.Authorizer) (Client, error) {
```

Comment:
This doesn't change anything but allow this interface to be mocked as a return value for testing.  The concrete type is still in use here as the return value for that is the concrete type which implements this interface.